EapWebInterface/src/main/java/de/hpi/unicorn/application/pages/input/generator/GeneratePanel.java

```diff
@@ -242,4 +305,111 @@ private void addSubmitButton() {
+    }
+
+    private void addImportSubmitButton() {
+        AjaxButton button = new AjaxButton("importValuesButton") {
+            @Override
+            public void onSubmit(AjaxRequestTarget target, Form importForm) {
+                final FileUpload uploadedFile = uploadField.getFileUpload();
+                if (uploadedFile == null) {
+                    GeneratePanel.this.page.getFeedbackPanel().error("File not found.");
+                    target.add(GeneratePanel.this.page.getFeedbackPanel());
+                    return;
+                }
+                // make sure provided file is json
+                final String fileName = uploadedFile.getClientFileName();
+                String fileFormat = fileName.substring(fileName.lastIndexOf('.') + 1);
+                if (!"json".equals(fileFormat)) {
+                    GeneratePanel.this.page.getFeedbackPanel().error("Please provide a json file.");
+                    target.add(GeneratePanel.this.page.getFeedbackPanel());
+                    return;
+                }
+                // generate inputs from file
+                File newFile;
+                Map<Object, Object> valueMap;
+                try {
+                    newFile = uploadedFile.writeToTempFile();
+                    String fileContent = Files.readFirstLine(newFile, Charset.defaultCharset());
+                    valueMap = JsonImporter.generateValuesFromString(fileContent);
+                    if (valueMap == null) {
+                        GeneratePanel.this.page.getFeedbackPanel().error("Values could not be read.");
+                        target.add(GeneratePanel.this.page.getFeedbackPanel());
+                        return;
+                    }
+                } catch (Exception e) {
+                    GeneratePanel.this.page.getFeedbackPanel().error("File could not be read.");
+                    target.add(GeneratePanel.this.page.getFeedbackPanel());
+                    return;
+                }
+
+                eventCountField.getModel().setObject((int) valueMap.get("eventCount"));
+                eventCountField.clearInput();
+                target.add(eventCountField);
+
+                scaleFactorField.getModel().setObject((int) valueMap.get("scaleFactor"));
+                scaleFactorField.clearInput();
+                target.add(scaleFactorField);
+
+                timestampField.getModel().setObject((String) valueMap.get("timestamp"));
+                timestampField.clearInput();
+                target.add(timestampField);
+
+                eventTypeDropDown.getModel().setObject((EapEventType) valueMap.get("eventType"));
+                eventTypeDropDown.clearInput();
+                target.add(eventTypeDropDown);
+
+                Map<TypeTreeNode, String> values = (Map<TypeTreeNode, String>) valueMap.get("values");
+                attributeInputs.clear();
+                for (TypeTreeNode inputValue : values.keySet()) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Iterate over the "entrySet" instead of the "keySet". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS2864)